vulcanai2/models/basenetwork.py

```diff
@@ -478,8 +515,13 @@ def save_model(self, save_path=None):
         os.makedirs(module_save_path)  # let this throw an error if it already exists
 
         # recursive recursive recursive
-        if self._input_network is not None:
-            self._input_network.save_model(save_path)
+        if self._input_networks is not None:
+            if self._input_network:
```

Comment:
you should probably loop through the `_input_networks` and save each of those first. We should revert to only using the `_input_networks` attribute instead of both that and `self_input_netowrk`